File: php/src/Api/IngestEndpointTransformation.php
<?php

// this file is @generated
declare(strict_types=1);

namespace Svix\Api;

use Svix\Exception\ApiException;
use Svix\Models\IngestEndpointTransformationOut;
use Svix\Models\IngestEndpointTransformationPatch;
use Svix\Request\SvixHttpClient;

class IngestEndpointTransformation
{
    public function __construct(
        private readonly SvixHttpClient $client,
    ) {
    }

    /**
     * Get the transformation code associated with this ingest endpoint.
     *
     * @throws ApiException
     */
    public function transformation(
        string $sourceId,
        string $endpointId,
    ): IngestEndpointTransformationOut {
        $request = $this->client->newReq('GET', "/ingest/api/v1/source/{$sourceId}/endpoint/{$endpointId}/transformation");
        $res = $this->client->send($request);

        return IngestEndpointTransformationOut::fromJson($res);
    }

    /**
     * Set or unset the transformation code associated with this ingest endpoint.
     *
     * @throws ApiException
     */
    public function patch(
        string $sourceId,
        string $endpointId,
        IngestEndpointTransformationPatch $ingestEndpointTransformationPatch,
    ): void {
        $request = $this->client->newReq('PATCH', "/ingest/api/v1/source/{$sourceId}/endpoint/{$endpointId}/transformation");
        $request->setBody(json_encode($ingestEndpointTransformationPatch));
        $res = $this->client->sendNoResponseBody($request);
    }
}

File: rust/src/api/ingest_endpoint_transformation.rs
// this file is @generated
use crate::{error::Result, models::*, Configuration};

pub struct IngestEndpointTransformation<'a> {
    cfg: &'a Configuration,
}

impl<'a> IngestEndpointTransformation<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Get the transformation code associated with this ingest endpoint.
    pub async fn transformation(
        &self,
        source_id: String,
        endpoint_id: String,
    ) -> Result<IngestEndpointTransformationOut> {
        crate::request::Request::new(
            http::Method::GET,
            "/ingest/api/v1/source/{source_id}/endpoint/{endpoint_id}/transformation",
        )
        .with_path_param("source_id", source_id)
        .with_path_param("endpoint_id", endpoint_id)
        .execute(self.cfg)
        .await
    }

    /// Set or unset the transformation code associated with this ingest
    /// endpoint.
    pub async fn patch(
        &self,
        source_id: String,
        endpoint_id: String,
        ingest_endpoint_transformation_patch: IngestEndpointTransformationPatch,
    ) -> Result<()> {
        crate::request::Request::new(
            http::Method::PATCH,
            "/ingest/api/v1/source/{source_id}/endpoint/{endpoint_id}/transformation",
        )
        .with_path_param("source_id", source_id)
        .with_path_param("endpoint_id", endpoint_id)
        .with_body_param(ingest_endpoint_transformation_patch)
        .returns_nothing()
        .execute(self.cfg)
        .await
    }
}
